API_SPECS_DEVICE_MODELS_V2.md

```diff
@@ -0,0 +1,459 @@
+          "MessageSchema": {
+            "Name": "truck-cargotemperature;v1",
+            "Format": "JSON",
+            "Fields": {
+              "cargotemperature": "Double",
+              "cargotemperature_unit": "Text"
+            }
+          }
+        }
+      ],
+      "CloudToDeviceMethods": {
+        "DecreaseCargoTemperature": {
+          "Type": "javascript",
+          "Path": "TBD.js"
+        },
+        "IncreaseCargoTemperature": {
+          "Type": "javascript",
+          "Path": "TBD.js"
+        }
+      },
+      "$metadata": {
+        "$type": "DeviceModel;1",
+        "$uri": "/v1/devicemodels/truck-01"
+      }
+    }
+  ],
+  "$metadata": {
+    "$type": "DeviceModelList;1",
+    "$uri": "/v1/devicemodels"
+  }
+}
+```
+
+### Get a device model by id
+
+Request:
+```
+GET /v2/devicemodels/${id}
+```
+
+Response example:
+```
+200 OK
+Content-Type: application/json; charset=utf-8
+```
+```json
+{
+    "ETag": "0000-0000-0000",
+    "Id": "00000000-000000-0000-0000-000000",
+    "Version": "0.0.0",
+    "Name": "Chiller",
+    "Description": "Chiller with external temperature, humidity and pressure sensors.",
+    "Protocol": "MQTT",
+    "Type": "",
+    "Simulation": {
+        "InitialState": {
+            "online": true,
+            "temperature": 75,
+            "temperature_unit": "F",
+            "humidity": 70,
+            "humidity_unit": "%",
+            "pressure": 150,
+            "pressure_unit": "psig",
+            "simulation_state": "normal_pressure"
+        },
+        "Interval": "00:00:10",
+        "Scripts": [
+            {
+                "Type": "javascript",
+                "Path": "chiller-01-state.js"
+            }
+        ]
+    },
+    "Properties": {},
+    "Telemetry": [
+        {
+            "Interval": "00:00:10",
+            "MessageTemplate": "{\"temperature\":${temperature},\"temperature_unit\":\"${temperature_unit}\",\"humidity\":${humidity},\"humidity_unit\":\"${humidity_unit}\",\"pressure\":${pressure},\"pressure_unit\":\"${pressure_unit}\"}",
+            "MessageSchema": {
+                "Name": "chiller-sensors;v1",
+                "Format": "JSON",
+                "Fields": {
+                    "temperature": "Double",
+                    "temperature_unit": "Text",
+                    "humidity": "Double",
+                    "humidity_unit": "Text",
+                    "pressure": "Double",
+                    "pressure_unit": "Text"
+                }
+            }
+        }
+    ],
+    "CloudToDeviceMethods": {},
+    "$metadata": {
+        "$type": "DeviceModel;2",
+        "$uri": "/v2/devicemodels/00000000-000000-0000-0000-000000",
+        "$created": "2018-03-29T23:33:03+00:00",
+        "$modified": "2018-03-29T23:33:03+00:00"
+    }
+}
+```
+
+## Creating device models
+
+### Creating stock device models
+
+To create a new stock device model, a new configuration file is added to the folder
```

Comment:
for "new configuration file is added to the folder where the configuration files are stored" could you also add the path or a more specific explanation of where that would be?